vmm/src/memory_manager.rs

```diff
@@ -980,13 +986,13 @@ impl MemoryManager {
     //
     // Calculate the start address of an area next to RAM.
     //
-    // If the next area is device space, there is no gap.
-    // If the next area is hotplugged RAM, the start address needs to be aligned
+    // If memory hotplug is not allowed, there is no gap.
+    // If memory hotplug is allowed, the start address needs to be aligned
     // to 128MiB boundary, and a gap of 256MiB need to be set before it.
     // On x86_64, it must also start at the 64bit start.
     #[allow(clippy::let_and_return)]
-    fn start_addr(mem_end: GuestAddress, with_gap: bool) -> GuestAddress {
-        let start_addr = if with_gap {
+    fn start_addr(mem_end: GuestAddress, allow_mem_hotplug: bool) -> GuestAddress {
+        let start_addr = if allow_mem_hotplug {
```

Comment:
Oh I see the problem you're trying to solve, but unfortunately, you're solving this only if we hotplug all the memory in one go. Let's say I start a VM with 4G, and 3G of hotpluggable memory. If I hotplug 1G at a time, I will never be able to reach the expected 7G, but instead it will be around 7G - (256M + 256M + 256M) + 256M. And this get even worse if we hotplug only 128M at a time :(

@rbradford I think one way to solve this issue would be to keep the alignment on 128MiB but remove the 256MiB gap. Because all extra regions would be aligned on 128MiB and they would be also 128MiB multiples in size, we wouldn't run into any gap issue in the address space. WDYT?

Reply:
@sboeuf You are right. With the current way of using `start_addr()`, we are reserving `256M` each time we are hotplugging memory. I forgot about that.

Reply:
As discussed offline, I will fix the issue related on reserving `256` gaps in a separate PR.